client/network/src/protocol/generic_proto/behaviour.rs

```diff
@@ -1291,6 +1329,9 @@ impl NetworkBehaviour for GenericProto {
 					let event = GenericProtoOut::CustomProtocolOpen { peer_id: source, received_handshake };
 					self.events.push_back(NetworkBehaviourAction::GenerateEvent(event));
 				} else {
+					if let Some(metrics) = &self.metrics {
+						metrics.non_primary_connections_opened.inc();
+					}
```

Comment:
I would strongly prefer to report that in `service.rs` with all the other metrics.
In my opinion we should when it is possible not infest the code with metrics and instead have metrics on the side of the main logic.